config/src/lua.rs

```diff
@@ -71,19 +71,36 @@ pub fn make_lua_context(config_file: &Path) -> anyhow::Result<Lua> {
                 }
             }
         }
+        let config_file_str = config_file
+            .to_str()
+            .ok_or_else(|| anyhow!("config file path is not UTF-8"))?;
 
-        wezterm_mod.set(
-            "config_file",
-            config_file
-                .to_str()
-                .ok_or_else(|| anyhow!("config file path is not UTF-8"))?,
-        )?;
+        lua.set_named_registry_value("wezterm-watch-paths", config_file_str)?;
+        wezterm_mod.set("config_file", config_file_str)?;
         wezterm_mod.set(
             "config_dir",
             config_dir
                 .to_str()
                 .ok_or_else(|| anyhow!("config dir path is not UTF-8"))?,
         )?;
+        wezterm_mod.set(
+            "add_to_config_reload_watch_list",
+            lua.create_function(|lua, args: Variadic<String>| {
+                let watch_path: String = lua.named_registry_value("wezterm-watch-paths")?;
+                log::debug!("watch_path: {}, args: {:?}", watch_path, args);
+                lua.set_named_registry_value(
+                    "wezterm-watch-paths",
+                    format!("{};{}", watch_path, args.join(";")),
+                )
+            })?,
+        )?;
+        wezterm_mod.set(
+            "set_config_reload_watch_list",
```

Comment:
I'm kinda leaning towards not including this; the way I see this working is purely additive.  The ability to set the complete list means that there is potential for one module to trample on things set by other modules, which feels bad if there isn't a corresponding getter function, and I think also adding *that* will have an outsize maintenance cost compared to the basic need to just add a path.

Reply:
I think that adding a getter does make sense. as I would want to be able to make it ignore the default file, especially performance-wise.